### DISCUSSION TITLE
Issue with Importing and Displaying DAG Code in Apache Airflow UI when use a subproject

### DISCUSSION THREAD

**Discussion by brunojus** [apache/airflow#35071] [Q&A]
I am facing a problem when importing DAGs into Apache Airflow using a script, as I am not able to see the DAG code in the Airflow UI. It only displays the code used for importation. Here's the code I'm using:

```
import os
from airflow.models import DagBag
import sys

dags_dirs = [
    '/opt/airflow/projetos/basevinculos_py',
    '/opt/airflow/projetos/pdi_consulta'
]

for dir in dags_dirs:
    sys.path.append(os.path.expanduser(dir))

for dir in dags_dirs:
    dag_bag = DagBag(os.path.expanduser(dir))

    if dag_bag:
        for dag_id, dag in dag_bag.dags.items():
            globals()[dag_id] = dag
```

I have verified that the DAGs are successfully imported into Airflow, but the Airflow UI does not display the code for the imported DAGs. I've ensured that the DAG definition files are correctly formatted and contain the necessary DAG structure. I'm using Apache Airflow 2.7.2. [Code-UI](https://i.stack.imgur.com/tzY0W.png)

The graph tab shows correctly the tasks flow correctly, but unfortunately in the code tab shows me only the code that I am using to deal with the DagBag.

Thanks in advance

**Comment by potiuk** [comment]
Yes. This is expected behaviour. Airflow code tab shows the code of the DAG file that has been parsed in order to import the DAGs. This is by design. If you want to have your DAGs shown in the UI, you must make them part of the DAGs folder so that DAG file processor parses and serializes them to the DB directly. Solving "general" problem of showing DAG code for all imported python code is not straightforward and easy, but if you have an idea how to do it and make a proposal - at the devlist or in PR, you are most welcome.
